src/carplay.cs

```diff
@@ -1008,4 +1289,556 @@ interface CPTemplateApplicationDashboardScene {
+	{
+		[Export ("initWithName:image:")]
+		IntPtr Constructor (string name, UIImage image);
+
+		[Export ("name")]
+		string Name { get; set; }
+
+		[Export ("image", ArgumentSemantic.Strong)]
+		UIImage Image { get; set; }
+
+		[NullAllowed, Export ("actions", ArgumentSemantic.Copy)]
+		CPButton[] Actions { get; set; }
+
+		[NullAllowed, Export ("subtitle")]
+		string Subtitle { get; set; }
+
+		[NullAllowed, Export ("informativeText")]
+		string InformativeText { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPButton))]
+	interface CPContactCallButton
+	{
+		[Export ("initWithImage:handler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (UIImage image, [NullAllowed] Action<CPButton> handler);
+
+		[Export ("initWithHandler:")]
+		IntPtr Constructor ([NullAllowed] Action<CPButton> handler);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPButton))]
+	interface CPContactDirectionsButton
+	{
+		[Export ("initWithImage:handler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (UIImage image, [NullAllowed] Action<CPButton> handler);
+
+		[Export ("initWithHandler:")]
+		IntPtr Constructor ([NullAllowed] Action<CPButton> handler);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPTemplate))]
+	[DisableDefaultCtor]
+	interface CPContactTemplate : CPBarButtonProviding
+	{
+		[Export ("initWithContact:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CPContact contact);
+
+		[Export ("contact", ArgumentSemantic.Strong)]
+		CPContact Contact { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CPInformationItem : NSSecureCoding
+	{
+		[Export ("initWithTitle:detail:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] string title, [NullAllowed] string detail);
+
+		[NullAllowed, Export ("title")]
+		string Title { get; }
+
+		[NullAllowed, Export ("detail")]
+		string Detail { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPTemplate))]
+	[DisableDefaultCtor]
+	interface CPInformationTemplate
+	{
+		[Export ("initWithTitle:layout:items:actions:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string title, CPInformationTemplateLayout layout, CPInformationItem[] items, CPTextButton[] actions);
+
+		[Export ("layout")]
+		CPInformationTemplateLayout Layout { get; }
+
+		[Export ("title")]
+		string Title { get; set; }
+
+		[Export ("items", ArgumentSemantic.Copy)]
+		CPInformationItem[] Items { get; set; }
+
+		[Export ("actions", ArgumentSemantic.Copy)]
+		CPTextButton[] Actions { get; set; }
+	}
+
+	// warning BI1115: bgen: The parameter 'completionHandler' in the delegate 'CarPlay.ListImageRowItemHandler' does not have a [CCallback] or [BlockCallback] attribute. Defaulting to [CCallback].
+	delegate void ListImageRowItemHandler (CPListImageRowItem item, nint index, Action completionHandler);
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (CPBaseListItem))]
```

Comment:
since the constructor `initWithText:images:` does not allow null `text` or `images` then it's likely `init` should not be used. Unless there's other evidence that it's available then add `[DisableDefaultCtor]` to the type.

Reply:
it's a weird one since the `Text` property allows null...
and it's not 100% clear (at least to me) if `images` == `gridImages` (looks likely)

Reply:
`extern NSUInteger const CPMaximumNumberOfGridImages;` -> likely `images` == `gridImages`

Added `[DisableDefaultCtor]` 👍